pkg/cmd/server/origin/run_components.go

```diff
@@ -376,7 +376,7 @@ func (c *MasterConfig) RunDeploymentImageChangeTriggerController() {
 // RunSDNController runs openshift-sdn if the said network plugin is provided
 func (c *MasterConfig) RunSDNController() {
 	oClient, kClient := c.SDNControllerClients()
-	controller, err := sdnfactory.NewMasterPlugin(c.Options.NetworkConfig.NetworkPluginName, oClient, kClient)
+	controller, err := sdnfactory.NewMasterPlugin(c.Options.NetworkConfig.NetworkPluginName, oClient, kClient, c.EtcdHelper)
```

Comment:
The controllers will not have etcd access very soon (possibly as soon as 3.3).  That's partially why we keep asking to keep this under the API server level.  I think any fix here has to be API only (from the controller) and the rest handled in the apiserver.